homeassistant/components/ultrasync/config_flow.py

```diff
@@ -0,0 +1,148 @@
+"""Config flow for Interlogix Ultrasync Hub."""
+import logging
+from typing import Any, Dict, Optional
+
+import ultrasync
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_NAME,
+    CONF_PIN,
+    CONF_SCAN_INTERVAL,
+    CONF_USERNAME,
+)
+from homeassistant.core import callback
+from homeassistant.helpers.typing import ConfigType, HomeAssistantType
+
+from .const import DEFAULT_NAME, DEFAULT_SCAN_INTERVAL
+from .const import DOMAIN  # pylint: disable=unused-import
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class AuthFailureException(IOError):
+    """A general exception we can use to track Authentication failures."""
+
+
+def validate_input(hass: HomeAssistantType, data: dict) -> Dict[str, Any]:
+    """Validate the user input allows us to connect."""
+
+    usync = ultrasync.UltraSync(
+        host=data[CONF_HOST], user=data[CONF_USERNAME], pin=data[CONF_PIN]
+    )
+
+    # validate by attempting to authenticate with our hub
+
+    if not usync.login():
+        # report our connection issue
+        raise AuthFailureException()
+
+    return True
+
+
+class UltraSyncConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """UltraSync config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    def __init__(self, config_entry):
+        """Initialize options flow."""
+        self.config_entry = config_entry
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry):
+        """Get the options flow for this handler."""
+        return UltraSyncOptionsFlowHandler(config_entry)
+
+    async def async_step_import(
```

Comment:
Why do you support import on new integration?

Reply:
@epenet thank you for your feedback!

In the latest push (just now) I've addressed all of your comments and dropped the YAML and import functionality. Just waiting to see the outcome of the upstream build :crossed_fingers: